From 93ff3829416785397148edd5ea611e1c9273c10a Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Mon, 7 Jul 2025 00:26:42 +0200
Subject: [PATCH 031/115] arm64: dts: mediatek: mt6785: Add smi support

Adding smi in the MT6785 dts will allow to add larbs, used by iommu.

Signed-off-by: Dinolek <git@dinolek.me>
---
 arch/arm64/boot/dts/mediatek/mt6785.dtsi | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt6785.dtsi b/arch/arm64/boot/dts/mediatek/mt6785.dtsi
index fcb40a975..bd4b84cf9 100644
--- a/arch/arm64/boot/dts/mediatek/mt6785.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt6785.dtsi
@@ -362,6 +362,7 @@ power-domain@MT6785_POWER_DOMAIN_AUDIO {
 				power-domain@MT6785_POWER_DOMAIN_CONN {
 					reg = <MT6785_POWER_DOMAIN_CONN>;
 					mediatek,infracfg = <&infracfg>;
+					mediatek,smi = <&smi_common>;
 					#power-domain-cells = <0>;
 				};
 
@@ -414,6 +415,7 @@ power-domain@MT6785_POWER_DOMAIN_MM {
 					clock-names = "mm";
 					mediatek,infracfg-nao = <&infracfg_nao>;
 					mediatek,infracfg = <&infracfg>;
+					mediatek,smi = <&smi_common>;
 					#address-cells = <1>;
 					#size-cells = <0>;
 					#power-domain-cells = <1>;
@@ -423,6 +425,7 @@ power-domain@MT6785_POWER_DOMAIN_CAM {
 						clocks = <&topckgen CLK_TOP_CAM>;
 						clock-names = "cam";
 						mediatek,infracfg = <&infracfg>;
+						mediatek,smi = <&smi_common>;
 						#power-domain-cells = <0>;
 					};
 
@@ -431,6 +434,7 @@ power-domain@MT6785_POWER_DOMAIN_ISP {
 						clocks = <&topckgen CLK_TOP_IMG>;
 						clock-names = "isp";
 						mediatek,infracfg = <&infracfg>;
+						mediatek,smi = <&smi_common>;
 						#power-domain-cells = <0>;
 					};
 
@@ -439,6 +443,7 @@ power-domain@MT6785_POWER_DOMAIN_VDEC {
 						clocks = <&topckgen CLK_TOP_VDEC>;
 						clock-names = "vdec";
 						mediatek,infracfg = <&infracfg>;
+						mediatek,smi = <&smi_common>;
 						#power-domain-cells = <0>;
 					};
 
@@ -447,6 +452,7 @@ power-domain@MT6785_POWER_DOMAIN_VENC {
 						clocks = <&topckgen CLK_TOP_VENC>;
 						clock-names = "venc";
 						mediatek,infracfg = <&infracfg>;
+						mediatek,smi = <&smi_common>;
 						#power-domain-cells = <0>;
 					};
 				};
@@ -517,6 +523,16 @@ mmsys: syscon@14000000 {
 			#reset-cells = <1>;
 		};
 
+		smi_common: smi@14019000 {
+			compatible = "mediatek,mt6785-smi-common",
+				     "mediatek,mt6779-smi-common";
+			reg = <0 0x14019000 0 0x1000>;
+			clocks = <&mmsys CLK_MM_SMI_COMMON>, <&mmsys CLK_MM_SMI_COMMON>,
+				 <&mmsys CLK_MM_GALS_COMM0>, <&mmsys CLK_MM_GALS_COMM1>;
+			clock-names = "apb", "smi", "gals0", "gals1";
+			power-domains = <&spm MT6785_POWER_DOMAIN_MM>;
+		};
+
 		imgsys: clock-controller@15020000 {
 			compatible = "mediatek,mt6785-imgsys", "syscon";
 			reg = <0 0x15020000 0 0x1000>;
-- 
2.56.0

